playwright/tests/syncTabs.spec.ts

```diff
@@ -0,0 +1,134 @@
+      await page.getByPlaceholder('Re-enter your password').fill(password)
+      await page.keyboard.press('Enter')
+      await expect(page.getByText('Loading account')).toBeVisible()
+      await expect(page.getByText('Loading account')).toBeHidden()
+      await expect(page.getByTestId('account-selector')).toBeVisible()
+      await expect(page).toHaveURL(new RegExp(`/account/${privateKeyAddress}`))
+
+      const tab2 = await context.newPage()
+      await testLockingIsSynced(page, tab2)
+      await expect(page.getByRole('button', { name: 'Unlock' })).toBeVisible()
+      await expect(tab2.getByRole('button', { name: 'Unlock' })).toBeVisible()
+    })
+
+    test('incognito', async ({ page, context }) => {
+      await addPersistedStorage(page)
+      await page.goto('/')
+      await page.getByRole('button', { name: 'Continue without the profile' }).click()
+      await expect(page.getByTestId('account-selector')).toBeHidden()
+
+      const tab2 = await context.newPage()
+      await tab2.goto('/')
+      await expect(tab2.getByRole('button', { name: 'Unlock' })).toBeHidden()
+      await tab2.goto('/open-wallet/private-key')
+      await tab2.getByPlaceholder('Enter your private key here').fill(privateKey)
+      await tab2.keyboard.press('Enter')
+      await expect(tab2.getByTestId('account-selector')).toBeVisible()
+      await expect(page.getByTestId('account-selector')).toBeVisible()
+
+      await testLockingIsSynced(page, tab2)
+      await expect(page.getByRole('button', { name: 'Unlock' })).toBeVisible()
+      await expect(tab2.getByRole('button', { name: 'Unlock' })).toBeVisible()
+    })
+
+    async function testLockingIsSynced(page: Page, tab2: Page) {
+      // Second tab should sync the opened wallet
+      await tab2.goto('/')
+      await expect(tab2.getByTestId('account-selector')).toBeVisible()
+      await tab2.getByRole('link', { name: 'Wallet' }).click()
+      await expect(tab2).toHaveURL(new RegExp(`/account/${privateKeyAddress}`))
+
+      // Second tab should not get stuck on loading after first tab closes wallet
+      await page.getByRole('button', { name: /(Close wallet)|(Lock profile)/ }).click()
+      await expect(page.getByText('Loading account')).toBeHidden()
+      await expect(page.getByTestId('account-selector')).toBeHidden()
+      await expect(tab2.getByText('Loading account')).toBeHidden()
+      await expect(tab2.getByTestId('account-selector')).toBeHidden()
+    }
+  })
+
+  test.describe('switch network and open second tab', () => {
+    test('unpersisted', async ({ page, context }) => {
+      await page.goto('/open-wallet/private-key')
+      await page.getByPlaceholder('Enter your private key here').fill(privateKey)
+      await page.keyboard.press('Enter')
+      await expect(page).toHaveURL(new RegExp(`/account/${privateKeyAddress}`))
+      const tab2 = await context.newPage()
+      await testSyncingNetwork(page, tab2)
+    })
```

Comment:
This is failing :/ and it's a mess

- first tab initializes
- attempts to sync state from other tabs
- networkSaga initializes, dispatches 'skipUnlocking', and fetches mainnet network info
- **user switches to testnet**, networkSaga fetches testnet network info
- second tab initializes
- second tab: attempts to sync state from other tabs
- second tab: networkSaga initializes, dispatches 'skipUnlocking', and fetches mainnet network info
- first tab: sends sync state, including testnet selection
- first tab: gets synced skipUnlocking, and fetches mainnet network info
- second tab: gets synced state, including testnet selection
- first tab: dispatches selectNetwork with fetched mainnet network info
- second tab: gets synced selectNetwork with fetched mainnet network info
- both tabs are **showing mainnet** instead of testnet

Reply:
https://github.com/oasisprotocol/oasis-wallet-web/pull/975/commits/c125984a415bd1a3fd9dd61c9ce6f2634e8ef8eb added a timeout to wait for tabs to sync, and only skipUnlocking if tabs don't sync